qemu/tests/qemu_guest_agent.py

```diff
@@ -777,6 +777,50 @@ def gagent_check_fsfreeze(self, test, params, env):
         # Finally, do something after thaw.
         self._action_after_fsthaw(test, params, env)
 
+    def gagent_check_read_write_guestfile(self, test, params, env):
+        """
+        Test guest agent commands "guest-file-open/write/read/close"
+
+        Test steps:
+        1) open file with mode "w+" in guest.
+        2) write "hello word" to file.
+        3) close file.
+        4) open file with mode "r+" in guest.
+        5) read file
+        6) close file
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        if not self.env or not self.params:
+            test.error("You should run 'setup' method before test")
+
+        if not (self.vm and self.vm.is_alive()):
+            vm = self.env.get_vm(self.params["main_vm"])
+            vm.verify_alive()
+            self.vm = vm
+
+        agent_file = "/tmp/agentfile"
+        #if file not exist,mode "w+" will create a new file
+        open_args_w = {"path": agent_file, "mode": "w+"}
+        ret = int(self.gagent.openfile(open_args_w))
+        write_str = "aGVsbG8gd29ybGQhCg=="
+        write_args = {"handle": ret, "buf-b64": write_str}
+        ret_write = self.gagent.writefile(write_args)
+        close_args_w = {"handle": ret}
+        self.gagent.closefile(close_args_w)
+
+        #mode r+ open an exist file only
+        open_args_r = {"path": agent_file, "mode": "r+"}
+        ret = int(self.gagent.openfile(open_args_r))
+        read_args = {"handle": ret, "count": 1024}
+        ret_read = self.gagent.readfile(read_args)
+        if not ret_read["buf-b64"] == write_str:
+            test.fail("Read the contents and files do not match")
+        close_args_r = {"handle": ret}
+        self.gagent.closefile(close_args_r)
```

Comment:
Checked the related function in https://github.com/avocado-framework/avocado-vt/pull/1116, use  success_resp=False as default.  I am puzzled if there has exception when open/close/read/write file ?

Reply:
@yanan-fu  Thanks,  I have updated this part in patch 1116